src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -48,430 +54,434 @@
+    private String region;
+    private TreeSet<AwsHardwareRequirements> flavors;
+    private LaunchCommandGenerator launchCommandGenerator;
+
+    public AwsV2ComputePlugin(String confFilePath) {
+        Properties properties = PropertiesUtil.readProperties(confFilePath);
+        this.region = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_REGION_SELECTION_KEY);
+        this.defaultSubnetId = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_DEFAULT_SUBNET_ID_KEY);
+        this.defaultGroupId = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_DEFAULT_SECURITY_GROUP_ID_KEY);
+        this.flavorsFilePath = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_FLAVORS_TYPES_FILE_PATH_KEY);
+        this.launchCommandGenerator = new DefaultLaunchCommandGenerator();
+        this.flavors = new TreeSet<AwsHardwareRequirements>();
+    }
+
+    @Override
+    public String requestInstance(ComputeOrder computeOrder, AwsV2User cloudUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.REQUESTING_INSTANCE_FROM_PROVIDER));
+        Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+        AwsHardwareRequirements flavor = findSmallestFlavor(computeOrder, cloudUser);
+        RunInstancesRequest request = buildRequestInstance(computeOrder, flavor);
+        return doRequestInstance(computeOrder, flavor, request, client);
+    }
+
+    @Override
+    public ComputeInstance getInstance(ComputeOrder computeOrder, AwsV2User cloudUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE_S, computeOrder.getInstanceId()));
+        Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+        updateHardwareRequirements(cloudUser);
+        String instanceId = computeOrder.getInstanceId();
+        return doGetInstance(instanceId, client);
+    }
+
+    @Override
+    public void deleteInstance(ComputeOrder computeOrder, AwsV2User cloudUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE_S, computeOrder.getInstanceId()));
+        Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+        String instanceId = computeOrder.getInstanceId();
+        doDeleteInstance(instanceId, client);
+    }
+
+    @Override
+    public boolean isReady(String instanceState) {
+        return AwsV2StateMapper.map(ResourceType.COMPUTE, instanceState).equals(InstanceState.READY);
+    }
+
+    @Override
+    public boolean hasFailed(String instanceState) {
+        return false;
+    }
+
+    protected void doDeleteInstance(String instanceId, Ec2Client client) throws UnexpectedException {
+        TerminateInstancesRequest request = TerminateInstancesRequest.builder()
+                .instanceIds(instanceId)
+                .build();
+        try {
+            client.terminateInstances(request);
+        } catch (Exception e) {
```

Comment:
this is a sdkException, right?

Reply:
check the same thing all over the methods please

Reply:
You're right, the exception to be caught on this line is a SdkException, but another occurrence on line 243, I believe it should not be changed because other exceptions can occur in the referenced block.